### ISSUE TITLE
Option to serialize Vec<T> into comma-separated list

### ISSUE THREAD

**Issue by Salzian** [boxdot/serde-url-params-rs#13]
This would be nice for APIs that require a comma-separated list for authorization scopes.

I was looking into the source code myself, but I only have limited knowledge of the inner workings of `serde`. Not sure where a Vec is being iterated in the custom `Serializer` of this crate.

**Comment by boxdot** [comment]
Do you an example of how it should be used with an expected output? We could make it directly into a test and implement it.

**Comment by Salzian** [comment]
Here's an example where I wanted to use this crate for generating an OAuth url for Google Calendars:

```rust
#[derive(Serialize)]
pub(crate) struct AuthorizationParameters<'a> {
    ...
    scope: Vec<&'a str>,
    ...
}
```

Actual
```
scope=openid&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcalendar.events.owned.readonly
```

Needed
```
scope=openid,https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcalendar.events.owned.readonly
```

**Comment by boxdot** [comment]
I added an example in #14. Does it solve you problem?

https://github.com/boxdot/serde-url-params-rs/pull/14/files#diff-fbd250710bab29e09d41e0c500aca98633826a0f71a8a354659bc27476756093

**Comment by Salzian** [comment]
Sorry for the delay, I was enjoying the rest of my holidays.

In my quick tinkering, I failed to implement this for my use case. After looking at it again at a later time, I noticed the usage of the _Itertools_ crate. Seems like the `format` functionality "is" experimental in std (`slice_concat_trait`). Actually it's stable but seems like they [forgot to remove the annotations](https://github.com/rust-lang/rust/issues/27747#issuecomment-1055635718).

Either way, with your guidance on how to operate the _serde_ crate in this example, I think I can either replicate the behaviour with some custom implementation or I consider using _Itertools_.

Thanks!

Not sure if I should close this issue _now_ or if you want to link-close it with #14.
